lib/solidus_affirm/callback_hook/base.rb

```diff
@@ -6,7 +6,7 @@ def authorize!(payment)
         authorized_affirm = Affirm::Charge.find(payment.response_code)
         payment.amount = authorized_affirm.amount / 100.0
         payment.save!
-        payment.order.next!
+        payment.order.next! unless payment.order.confirm?
```

Comment:
We are assuming the step after `payment` is always `confirm` but this is not true for every store. Some could have a custom checkout flow and this would not work. 

What about calling `next` only if it's currently on `payment`?

Reply:
Yes of course, this makes sense. 👍